src/Network/Network/NetworkWatcher/ConnectionMonitor/SetAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -17,29 +17,52 @@
 using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
 using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
 using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 using System.Management.Automation;
 using MNM = Microsoft.Azure.Management.Network.Models;
 
 namespace Microsoft.Azure.Commands.Network
 {
-    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkWatcherConnectionMonitor", SupportsShouldProcess = true, DefaultParameterSetName = "SetByName"),OutputType(typeof(PSConnectionMonitorResult))]
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkWatcherConnectionMonitor", SupportsShouldProcess = true, DefaultParameterSetName = "SetByName"),
+                                                                       OutputType(typeof(PSConnectionMonitorResultV1)),
+                                                                       OutputType(typeof(PSConnectionMonitorResultV2))]
+
     public class SetAzureNetworkWatcherConnectionMonitorCommand : ConnectionMonitorBaseCmdlet
     {
         [Parameter(
              Mandatory = true,
              ValueFromPipeline = true,
              HelpMessage = "The network watcher resource.",
              ParameterSetName = "SetByResource")]
+        [Parameter(
+             Mandatory = true,
+             ValueFromPipeline = true,
+             HelpMessage = "The network watcher resource.",
+             ParameterSetName = "SetByConnectionMonitorV1")]
```

Comment:
What differentiates these two parameters?

Reply:
CMv2:
TestGroups, Outputs, Note

CMv1:
all other parameter (SourceResourceId, SourcePort, DestinationAddress, DestinationPort, DestinationResourceId)


so here, we create two parameter sets; one for CMV1 and one for CMv2

Reply:
Done what you suggested below